src/Compilers/CSharp/Test/Syntax/Parsing/NullConditionalAssignmentParsingTests.cs

```diff
@@ -0,0 +1,595 @@
+                        {
+                            N(SyntaxKind.OpenBracketToken);
+                            N(SyntaxKind.Argument);
+                            {
+                                N(SyntaxKind.IdentifierName);
+                                {
+                                    N(SyntaxKind.IdentifierToken, "b");
+                                }
+                            }
+                            N(SyntaxKind.CloseBracketToken);
+                        }
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Theory]
+        [MemberData(nameof(AllTestOptions))]
+        public void NullCoalescing_LeftMemberAccess(CSharpParseOptions parseOptions)
+        {
+            string source = "a?.b = c ?? d";
+            UsingExpression(source, parseOptions);
+
+            N(SyntaxKind.ConditionalAccessExpression);
+            {
+                N(SyntaxKind.IdentifierName);
+                {
+                    N(SyntaxKind.IdentifierToken, "a");
+                }
+                N(SyntaxKind.QuestionToken);
+                N(SyntaxKind.SimpleAssignmentExpression);
+                {
+                    N(SyntaxKind.MemberBindingExpression);
+                    {
+                        N(SyntaxKind.DotToken);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "b");
+                        }
+                    }
+                    N(SyntaxKind.EqualsToken);
+                    N(SyntaxKind.CoalesceExpression);
+                    {
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "c");
+                        }
+                        N(SyntaxKind.QuestionQuestionToken);
+                        N(SyntaxKind.IdentifierName);
+                        {
+                            N(SyntaxKind.IdentifierToken, "d");
+                        }
+                    }
+                }
+            }
+            EOF();
+        }
+
+        [Theory]
+        [InlineData(SyntaxKind.BarEqualsToken)]
+        [InlineData(SyntaxKind.AmpersandEqualsToken)]
+        [InlineData(SyntaxKind.CaretEqualsToken)]
+        [InlineData(SyntaxKind.LessThanLessThanEqualsToken)]
+        [InlineData(SyntaxKind.GreaterThanGreaterThanEqualsToken)]
+        [InlineData(SyntaxKind.GreaterThanGreaterThanGreaterThanEqualsToken)]
+        [InlineData(SyntaxKind.PlusEqualsToken)]
+        [InlineData(SyntaxKind.MinusEqualsToken)]
+        [InlineData(SyntaxKind.AsteriskEqualsToken)]
+        [InlineData(SyntaxKind.SlashEqualsToken)]
+        [InlineData(SyntaxKind.PercentEqualsToken)]
+        [InlineData(SyntaxKind.EqualsToken)]
+        [InlineData(SyntaxKind.QuestionQuestionEqualsToken)]
+        public void VariousAssignmentKinds_LeftMemberAccess(SyntaxKind kind)
```

Comment:
I didn't feel like setting up the combinatorial data for all the syntax kinds and language versions here so just kept the local functions approach.